src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -3588,6 +3596,11 @@ else if (split[0].equalsIgnoreCase("town"))
 				continue;
 
 			if (townBlockOwner instanceof Town && !townBlock.hasResident()) {
+				try {
+					BukkitTools.ifCancelledThenThrow(new TownBlockPermissionChangeEvent(townBlock, Arrays.toString(StringMgmt.remFirstArg(split))));
+				} catch (CancelledEventException e) {
+					sender.sendMessage(e.getCancelMessage());
+				}

```

Comment:
This looks like it has to be pulled upwards to line 3597, or we wont be throwing the event when a resident is using /res set perm.

Also, have you tested what `Arrays.toString(StringMgmt.remFirstArg(split)))` does when someone has potentially used more arguments than necessary and fills up `split` with garbage?

Reply:
I've changed the event to use a `TownyPermissionChange` that way this can be avoided. I've also added an additional check to prevent players from adding garbage to the `split`